openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -597,6 +618,301 @@ Location: https://client.example.net/cb?
+     The Wallet MUST NOT modify the URI in any way and treat it as an opaque value.
+     The Issuer MUST verify that the redirect URI in the `interactive_binding_token` parameter is correct, i.e., matches the one sent in response to the request to the `response_uri`.
+     Since the redirect URI MUST include a fresh, cryptographically random value, this check helps to prevent Session Fixation attacks, see (#iar-security).
+  3. The URL of the Interactive Authorization Request endpoint becomes the Origin for the request; i.e., the Wallet MUST ensure that `expected_origins` contains the Interactive Authorization Request endpoint URL.
+  4. For both signed and unsigned requests, the audience in the response (for example, the `aud` value in a Key Binding JWT) MUST be the Interactive Authorization Request, prefixed with `iar:`, for example `iar:https://example.com/iar`. A response containing a different audience value MUST NOT be accepted.
+
+#### Redirect to Web
+
+If the type is `redirect_to_web`, the Authorization Server wants to fall back to a regular interaction with the user.
+
+In this case, the Authorization server MUST include the key `request_uri` in the response.
+The Wallet MUST use the `request_uri` value to build an Authorization Request as defined in Section 4 of [@!RFC9126] and complete the rest of the authorization process as defined there.
+The Authorization Server MAY include the `expires_in` key as defined in [@!RFC9126].
+
+Non-normative Example:
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "require_interaction",
+  "type": "redirect_to_web",
+  "request_uri": "urn:ietf:params:oauth:request_uri:6esc_11ACC5bwc014ltc14eY22c",
+  "expires_in": 60
+}
+```
+
+#### Custom Interaction Extensions {#iar-custom-extensions}
+
+Additional, custom types of interactions MAY be defined by extensions of this specification to enable other types of interactions, for example, by interacting with a smart card.
+It is RECOMMENDED to use this extension point instead of modifying the OAuth protocol in order to facilitate interactions that require interactions with native components of the Wallet application.
+See (#iar-security) for additional security considerations.
+
+In the following non-normative example, this extension point is used to read the Betelgeuse Intergalactic ID card through an NFC interface in the Wallet. A token called `biic_token` is used to start the process.
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "require_interaction",
+  "type": "betelgeuse_intergalactic_id_card",
+  "biic_token": "73475cb40a568e8da8a045ced110137e159f890ac4da883b6b17dc651b3a8049"
+}
+```
+
+#### Preventing Session Fixation Attacks {#iar-security}
```

Comment:
Should this section also consider collusion between two Wallets? I feel that credentials use case lends itself to this sort of behaviour outlined below: 

For example let's say Dave wanted to obtain a AWS Certification that he did not have, but Alice does. Dave could collaborate with Alice (say for payment)

1. Dave initiates authentication
2. Dave passes the presentation request and nonce to Alice who share's her ID
3. She sends Dave the redirect_uri and completes authentication 
4. Issuer sends Dave the certification credential bound to a key controlled by Dave.